keepassxc-browser/content/keepassxc-browser.js

```diff
@@ -513,6 +513,19 @@ kpxc.prepareCredentials = async function() {
 
     kpxc.initLoginPopup();
     kpxc.initAutocomplete();
+
+    if (kpxc.settings.autoFillSingleEntry) {
+        const pageUuid = await sendMessage('page_get_login_id');
+        if (pageUuid) {
+            kpxc.credentials.some(c => {
```

Comment:
This is very wrong, you are using some,  a collector function,  to iterate. You need to use a for loop instead so that your return call actually ends the loop.

Reply:
> This is very wrong, you are using some, a collector function, to iterate. You need to use a for loop instead so that your return call actually ends the loop.

According to mdn, does it immediately end loop if some return true ?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some#return_value

Reply:
Absolutely not. `some` means collect certain elements that match certain criteria. Returning true means "include this considered element".

Reply:
Interesting, so this is a tester not a collector. Sigh...